agent/acs/handler/payload_handler.go

```diff
@@ -325,8 +325,11 @@ func isTaskStatusNotStopped(status api.TaskStatus) bool {
 // handleUnrecognizedTask handles unrecognized tasks by sending 'stopped' with
 // a suitable reason to the backend
 func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs.Task, err error, payload *ecsacs.PayloadMessage) {
+	seelog.Warnf("Received unexpected acs message, messageID: %s, task: %v, err: %v",
+		aws.StringValue(payload.MessageId), aws.StringValue(task.Arn), err)
+
 	if task.Arn == nil {
```

Comment:
Let's also modify this to be `if aws.StringValue(task.Arn) == ""`. That is a more complete check, that'll check both for the field not being set (`nil`) and set to an empty string.